### PR TITLE
db/datastates-cursor

### PR DESCRIPTION
**Pull request by bre1470**
- Added DataStates sync demo
- Merged in master/datastates
- Fixed call stack issue

### CODE REVIEW

File: samples/dashboards/demos/datastates-sync/demo.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        yAxis: {
+            title: {
+                enabled: false
+            }
+        }
+    };
+}
+
+// Build table with Highcharts.Series aliases
+function buildVegeTable() {
```
Comment (TorsteinHonsi) [inline samples/dashboards/demos/datastates-sync/demo.js:106]: `buildVegeTable` 😂

File: ts/Data/DataStates.ts [hunk 2]
```diff
@@ -0,0 +1,657 @@
...
+        return this;
+    }
+
+    /**
+     * Removes a lasting state cursor.
+     *
+     * @function #remitCursor
+     *
+     * @param {string} tableId
+     * The related table of the cursor.
```
Comment (karolkolodziej) [inline ts/Data/DataStates.ts:259]: Here should be probably table ID.

File: samples/dashboards/demos/datastates-sync/demo.js (hunk 1 above)
Comment (karolkolodziej) [reply samples/dashboards/demos/datastates-sync/demo.js:106]: catchy!

File: samples/dashboards/demos/datastates-sync/demo.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+function buildChartOptions(type, table, states) {
+    return {
+        chart: {
+            events: {
+                load: function () {
+                    const chart = this;
+                    const series = chart.series[0];
+
+                    // react to table states
+                    states.addListener(table.id, 'point.mouseOver', function (e) {
```
Comment (karolkolodziej) [inline samples/dashboards/demos/datastates-sync/demo.js:51]: Is this going to be replaced later by the HighchartsCompoment implementation?

**Review by karolkolodziej — APPROVED** [review]

**Review by goransle — APPROVED** [review]
Looks good!

**Review by pawellysy — APPROVED** [review]

**Pull request merged**